testcases/OpTestLPM.py

```diff
@@ -78,23 +100,59 @@ def lpm_setup(self):
             if cf.exitcode == 3:
                 self.firewall_status = False
         if self.firewall_status:
-             self.cv_HOST.host_run_command("systemctl stop firewalld.service")
+            self.cv_HOST.host_run_command("systemctl stop firewalld.service")
         rc = self.cv_HOST.host_run_command("lssrc -a | grep 'rsct \| rsct_rm'")
         if "inoperative" in str(rc):
             self.cv_HOST.host_run_command("startsrc -g rsct_rm; startsrc -g rsct")
             rc = self.cv_HOST.host_run_command("lssrc -a")
             if "inoperative" in str(rc):
                 raise OpTestError("LPM cannot continue as some of rsct services are not active")
 
+    def vnic_options(self, remote=''):
+        '''
+        Form the vnic_mappings param based on the adapters' details
+        provided.
+        '''
+        if remote:
+            cmd = []
+            for index in range(0, len(self.adapters)):
+                l_cmd = []
+                for param in [self.slot_num, 'ded', self.src_lpar_vios[index],
+                              self.vios_id[index], self.adapter_id[index],
+                              self.ports[index], self.bandwidth,
+                              self.target_adapter_id[index],
+                              self.target_ports[index]]:
+                    l_cmd.append(param)
+                cmd.append("/".join(l_cmd))
+        else:
+            cmd = []
+            for index in range(0, len(self.adapters)):
+                l_cmd = []
+                for param in [self.slot_num, 'ded',
+                              self.dest_lpar_vios[index],
+                              self.target_vios_id[index],
+                              self.target_adapter_id[index],
+                              self.target_ports[index], self.bandwidth,
+                              self.adapter_id[index], self.ports[index]]:
+                    l_cmd.append(param)
+                cmd.append("/".join(l_cmd))
+
+        return " -i \"vnic_mappings=\\\"%s\\\"\" " % ",".join(cmd)
+
     def lpar_migrate_test(self):
         self.check_pkg_installation()
         self.lpm_setup()
-        if not self.cv_HMC.migrate_lpar(self.src_mg_sys, self.dest_mg_sys):
+        cmd = ''
+        if self.slot_num:
+            cmd = self.vnic_options()
+        if not self.cv_HMC.migrate_lpar(self.src_mg_sys, self.dest_mg_sys, self.options, cmd):
```

Comment:
If you have some slot_num as a paramter, then cmd will have atlease "-i vnic_mappings" even if the slot or adapter not found.  Will the migrate fail, with that additional parameter with out any values .